tests/common/mod.rs

```diff
@@ -0,0 +1,279 @@
+    let bitcoind = BitcoinD::from_downloaded_with_conf(&conf).unwrap();
+
+    // Mine 101 blocks in our little regtest network so that the funds are spendable.
+    // (See https://bitcoin.stackexchange.com/questions/1991/what-is-the-block-maturation-time)
+    let address = bitcoind
+        .client
+        .get_new_address(None, Some(json::AddressType::Bech32))
+        .unwrap();
+    let address = address.require_network(Network::Regtest).unwrap();
+    bitcoind.client.generate_to_address(101, &address).unwrap();
+
+    (bitcoind, bitcoind_dir)
+}
+
+// setup_electrs sets up the electrs instance required (for the time being) for us to connect to an LDK node.
+pub async fn setup_electrs() -> ElectrsD {
+    let bitcoind_exe = env::var("BITCOIND_EXE")
+        .ok()
+        .or_else(|| electrsd::bitcoind::downloaded_exe_path().ok())
+        .expect(
+            "you need to provide an env var BITCOIND_EXE or specify a bitcoind version feature",
+        );
+    let bitcoind_dir = tempdir().unwrap();
+    let bitcoind_dir_path = bitcoind_dir.path().clone().to_path_buf();
+    let mut bitcoind_conf = electrsd::bitcoind::Conf::default();
+    let port = bitcoind::get_available_port().unwrap();
+    let zmq_block_port = bitcoind::get_available_port().unwrap();
+    let zmq_tx_port = bitcoind::get_available_port().unwrap();
+    let port_str = format!("-port={}", port);
+    let bind_str = format!("-bind=127.0.0.1:{}=onion", port);
+    let zmq_block_str = format!("-zmqpubrawblock=tcp://127.0.0.1:{}", zmq_block_port);
+    let zmq_tx_str = format!("-zmqpubrawtx=tcp://127.0.0.1:{}", zmq_tx_port);
+    bitcoind_conf.tmpdir = Some(bitcoind_dir_path);
+    bitcoind_conf.p2p = electrsd::bitcoind::P2P::Yes;
+    bitcoind_conf.args = vec![
+        "-server",
+        "-regtest",
+        &port_str,
+        &bind_str,
+        &zmq_block_str,
+        &zmq_tx_str,
+    ];
+    let bitcoind = electrsd::bitcoind::BitcoinD::with_conf(bitcoind_exe, &bitcoind_conf).unwrap();
+
+    let electrs_exe =
+        electrsd::downloaded_exe_path().expect("electrs version feature must be enabled");
+    let mut electrsd_conf = electrsd::Conf::default();
+    let electrsd_dir = tempdir().unwrap();
+    let electrsd_dir_path = electrsd_dir.path().clone().to_path_buf();
+    electrsd_conf.tmpdir = Some(electrsd_dir_path);
+    electrsd_conf.http_enabled = true;
+    electrsd_conf.network = "regtest";
+
+    ElectrsD::with_conf(electrs_exe, &bitcoind, &electrsd_conf).unwrap()
+}
+
+#[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-linux-amd64-v{}-beta.tar.gz", &LND_VERSION)
+}
+
+#[cfg(all(target_os = "macos", target_arch = "x86_64"))]
+fn lnd_download_filename() -> String {
+    format!("lnd-darwin-arm64-v{}-beta.tar.gz", &LND_VERSION)
+}
+
+#[cfg(all(target_os = "windows"))]
+fn lnd_download_filename() -> String {
+    panic!("Running integration tests on Windows os is not currently supported");
+}
```

Comment:
Why panic here vs just don't add the `target_os`? Then we won't compile / saves us some time.